### ISSUE TITLE
[java] UnnecessaryLocalBeforeReturn false positive when negating variable

### ISSUE THREAD

**Issue by C-Otto** [pmd/pmd#2206]
Rule Set: design

Code Sample demonstrating the issue:

```
public boolean isOk() {
    boolean isWrong = foo();
    return !isWrong;
}
```

I'd like to make the negation explicit, as the tiny character `!` is easy to read over.

Violation:

[UnnecessaryLocalBeforeReturn] Consider simply returning the value vs storing it in local variable 'isWrong'

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
